src/documents/ContextAwareness.md

```diff
@@ -0,0 +1,327 @@
+the "demo uploader" which mocks all of the engine-side implementation, and secondly the uploader can be run in various styles of integration tests which only mock the transport level which
+actually performs the file upload.
+
+### Example of dynamically broadcasting a fresh adaptation
+
+Finally, a third dimension of adaptation is supported by the Uploader's capability to be configured in a way that it will respond to previous instances of its own API - in particular that
+delivered for Infusion 1.2, released in April 2010, and Infusion 1.3, released in November 2010. This is implemented by allowing a dynamic contribution of a fresh dimension to the uploader's
+`contextAwareness` record from separate implementation files:
+
+```javascript
+fluid.defaults("fluid.uploader.compatibility.distributor.1_3", {
+    distributeOptions: {
+        record: {
+            "1_2": {
+                contextValue: "{fluid.uploader.requiredApi}.options.value",
+                equals: "fluid_1_2",
+                gradeNames: "fluid.uploader.compatibility.1_2"
+            }
+        },
+        target: "{/ fluid.uploader}.options.contextAwareness.apiCompatibility.checks"
+    }
+});
+
+// Actually construct a component instance performing the options broadcast into all uploaders
+fluid.constructSingle([], {
+    singleRootType: "fluid.uploader.compatibility.distributor",
+    type: "fluid.uploader.compatibility.distributor.1_3"
+});
+
+// The grade that contextAwareness ends up contributing into the uploader, if the rule is activated
+fluid.defaults("fluid.uploader.compatibility.1_2", {
+    transformOptions: {
+        transformer: "fluid.model.transformWithRules",
+        config: fluid.compat.fluid_1_2.uploader.optionsRules
+    }
+});
+```
+
+The defaults block `fluid.uploader.compatibility.distributor.1_3` contains an options distribution which appeals to the existence of a third dimension in the uploader's `contextAwareness`, named
+`apiCompatibility` &#8212; this can be done simply by arranging to broadcast the appropriate options into it. Once we have defined this options distribution, we actually need to construct a component
+a component instance which holds and operates them &#8212; this is done via the `fluid.constructSingle` line. This utility automatically arranges for a singleton instance, uniquified at the 
+component tree's top level by the type `singleRootType` which has a very similar function to the option of the same name consumed by the `fluid.resolveRootSingle` grade described in
+the documentation on [Contexts](Contexts.md#global-components-fluid.resolveRoot-and-fluid.resolveRootSingle).
+
+Having shown the basic operation of the _receiver_ of contextual information, we'll now describe the group of utilities, including `fluid.constructSingle` that we just met, which can be
```

Comment:
It would be better to lay out the idea that there are these two different types of parts of this API up front (i.e. near the beginning of the document), and then describe them in detail in subsequent sections. It would help the user get a picture of the broader context within which the details fit.